### PR TITLE
submit ubs data science task---ismet ferdi sen

### PR DESCRIPTION
**Pull request by steinbock277**
Dear Pavlin

Thank you for giving me chance to show my skills.

Best regards,
Ferdi

### CODE REVIEW

File: solution/task2_pca.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+# Convert to dataframe train data
+component_names = [f'PC{i+1}' for i in range(X_pca.shape[1])]
+X_pca = pd.DataFrame(X_pca, columns=component_names)
+
+# Convert to dataframe valid data
+component_names = [f'PC{i+1}' for i in range(X_valid_pca.shape[1])]
+X_valid_pca = pd.DataFrame(X_valid_pca, columns=component_names)
+
+#Calculate MI Scores
+mi_scores_pca = make_mi_scores(X_pca, y_train, discrete_features=False)
```
Comment (steinbock277) [inline solution/task2_pca.py]: I forgot hear to delete third parameter "discrete_features=False". Please delete it before running the code.

File: solution/task1.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+classifiers = [
+    SVC(kernel="linear", C=0.025),
+    SVC(),
+    DecisionTreeClassifier(max_depth=5),
+    RandomForestClassifier(max_depth=5, n_estimators=100, max_features=1),
+    #to avoid overfitting define early_stopping
+    MLPClassifier(hidden_layer_sizes=(100), alpha=1, max_iter=1000, early_stopping=True),
+    AdaBoostClassifier(),
+    GaussianNB()
+]
```
Comment (steinbock277) [inline solution/task1.py:68]: I did not define a **"random_state"** parameter for some of the classifiers which accept it like DecisionTreeClassifier, RandomForestClassifier and MLPClassifier. So on every iterations results can vary slightly.

File: solution/task2_pca.py (hunk 1 above)
Comment (pmavrodiev) [reply solution/task2_pca.py]: Can you do it yourself and submit again?

File: solution/task1.py (hunk 2 above)
Comment (pmavrodiev) [reply solution/task1.py:68]: Go ahead and define the random state. Then push again.

File: solution/task2_pca.py (hunk 1 above)
Comment (steinbock277) [reply solution/task2_pca.py]: It's done!

File: solution/task1.py (hunk 2 above)
Comment (steinbock277) [reply solution/task1.py:68]: Done!

**Pull request merged**